builtin/providers/digitalocean/resource_digitalocean_droplet.go

```diff
@@ -356,11 +351,8 @@ func resourceDigitalOceanDropletUpdate(d *schema.ResourceData, meta interface{})
 				"Error resizing droplet (%s): %s", d.Id(), err)
 		}
 
-		// Wait for the size to change
-		_, err = WaitForDropletAttribute(
-			d, newSize.(string), []string{"", oldSize.(string)}, "size", meta)
-
-		if err != nil {
+		// Wait for the resize action to complete.
+		if err := waitForAction(client, action); err != nil {
```

Comment:
I changed `WaitForDropletAttribute` to `waitForAction`, because we don't know what disk size value to wait for, as it's not available from a resource configuration – all we've got is a `size` value, but it represents RAM size, not disk size.

So the only way is to wait for a resize action to complete.